radixdlt-java/src/test/java/com/radixdlt/client/application/translate/TokenBalanceReducerTest.java

```diff
@@ -23,7 +24,43 @@ public void testSimpleBalance() {
 		when(ownedTokensParticle.getTokenClassReference()).thenReturn(token);
 
 		TokenBalanceReducer reducer = new TokenBalanceReducer();
-		TokenBalanceState tokenBalance = reducer.reduce(new TokenBalanceState(), SpunParticle.up(ownedTokensParticle));
+		TokenBalanceState tokenBalance = reducer.reduce(new TokenBalanceState(), TransitionedParticle.n2u(ownedTokensParticle));
 		assertThat(tokenBalance.getBalance().get(token).getAmount().compareTo(TokenClassReference.subUnitsToDecimal(10L))).isEqualTo(0);
 	}
+
+	@Test
+	public void transitionAndRevertTest() {
+		OwnedTokensParticle ownedTokensParticle = mock(OwnedTokensParticle.class);
+		RadixHash hash = mock(RadixHash.class);
+		when(ownedTokensParticle.getAmount()).thenReturn(10L);
+		when(ownedTokensParticle.getHash()).thenReturn(hash);
+		when(ownedTokensParticle.getDson()).thenReturn(new byte[] {1});
+		TokenClassReference token = mock(TokenClassReference.class);
+		when(ownedTokensParticle.getTokenClassReference()).thenReturn(token);
+
+		TokenBalanceReducer reducer = new TokenBalanceReducer();
+		TokenBalanceState tokenBalance0 = reducer.reduce(new TokenBalanceState(), TransitionedParticle.n2u(ownedTokensParticle));
+		TokenBalanceState tokenBalance1 = reducer.reduce(tokenBalance0, TransitionedParticle.u2n(ownedTokensParticle));
```

Comment:
We might want to consider spelling `TransitionedParticle.u2n` and its variants out. Even though I know what `u2n `means, I find it a bit hard to read at a glance.